### DISCUSSION TITLE
Close layer and navigate

### DISCUSSION THREAD

**Discussion by alexandremjacques** [unpoly/unpoly#637] [Q&A] [answered]
Hi.

If the user is on a new layer (origined from the root layer) and clicks a link that is supposed to dismiss the current layer and navigate to another page in the main target. What would be the best approach?

I'm trying to `up-target` the naviagation link to the `:main` selector but, since the current layer also has a `up-main`, it doesn't work. 

TIA

**Comment by adam12** [comment, marked as answer]
There are a [few options](https://unpoly.com/closing-overlays). I think it will depend on what kind of interaction you are doing. I might be inclined to use `[up-accept]` or `[up-dismiss]` but setting a close condition is generally better since you're assuming the content is aware it's running in an overlay, which isn't always true.

This is done when opening the layer with `[up-accept-location]` and similar.

**Reply by alexandremjacques** [reply]
Since the interaction was a simple `a href` link to a details page, `up-layer="parent"` (`parent` being the root layer in this case) did the trick.

Thanks again.
